### PR TITLE
Feature/a11y

### PR DESCRIPTION
**Pull request by vbence86**
#### Intent
To integrate accessibility test to metal-clay-components.

### CODE REVIEW

**Comment by carloslancha** [comment]
Don't forget to add `role="radiogroup"` to the `ClayRadio` wrapper in https://github.com/metal/metal-clay-components/pull/97/commits/181f064bcb978cb006000afcb270eda2fb50ca26#diff-51e59515af5c50ecfe051746bc4b243bR29 ;)

**Comment by vbence86** [comment]
@carloslancha that's the thing, it needs to be added to a container element that incorporates all elements with the same `name` attribute. This needs to be controlled either in the Demo or a prospective clay component that is responsible for containing ClayRadios.

**Comment by carloslancha** [comment]
@vbence86 Ah ok! It's only when radios share the same name! Isn't it? I didn't get it right before, sorry? 😅

File: packages/clay-button/src/ClayButton.soy [hunk 1]
```diff
@@ -135,8 +129,12 @@
 
 	{if $ariaLabel}
 		aria-label="{$ariaLabel}"
+	{elseif $label}
```
Comment (carloslancha) [inline packages/clay-button/src/ClayButton.soy:141]: You need to add `label` and `icon` as parameters of `.attributes` template and pass'em from `.render` (That's why compilation is failing)

Comment (vbence86) [reply packages/clay-button/src/ClayButton.soy:141]: Yes, I got a failing build and I've amended this by now, thanks for the head-ups though.

**Comment by vbence86** [comment]
Yes, individual radio buttons with unique `name` attribute pass a11y with no need of any additional aria tags.

**Comment by vbence86** [comment]
a11y module looks good now. We've got few violations captured with regards to the newly added ClayDropDown. I'm fixing them now.

captura de pantalla 2017-09-27 a las 9 23 20

**Comment by jbalsas** [comment]
@matuzalemsteles, @carloslancha, can you review the changes here?

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
LGTM! Good job!!

File: packages/clay-dropdown/src/ClayDropdownBase.soy [hunk 2]
```diff
@@ -135,7 +134,7 @@
 		"
 	{/let}
 
-	<ul {$listAttributes}>
+	<div {$listAttributes}>
```
Comment (jbalsas) [inline packages/clay-dropdown/src/ClayDropdownBase.soy]: why are we moving away from semantic tags?

Comment (vbence86) [reply packages/clay-dropdown/src/ClayDropdownBase.soy]: `<ul>` elements on no account must incorporate elements other than `<li>`, `<script>`, `<template>`. (reference - https://dequeuniversity.com/rules/axe/1.1/list)

A solution would be to append all content wrapped by `<li>` elements, but due the code there seems to wrap everything in `<div>` I thought I'd follow that approach and remove `<ul>` completely. This change could likely imply unexpected design consequences though.

Comment (jbalsas) [reply packages/clay-dropdown/src/ClayDropdownBase.soy]: If this is a list of items, it needs to be a list. If an item is not an item, either it needs to be put outside the list or the list needs to be declared deeper down to enclose the real list of items.

Comment (vbence86) [reply packages/clay-dropdown/src/ClayDropdownBase.soy]: Agreed. With this in mind, I will have to refactor the whole dropdown soy file.

Comment (jbalsas) [reply packages/clay-dropdown/src/ClayDropdownBase.soy]: 👍

Comment (carloslancha) [reply packages/clay-dropdown/src/ClayDropdownBase.soy]: I think making all `{deltemplate ClayDropdown.item}` and variants `li` elements (and `ul` the parent) and adding `aria-hidden` and `role=presentation` to the `separator` and `header` variants should do the work.

Comment (vbence86) [reply packages/clay-dropdown/src/ClayDropdownBase.soy]: I'm addressing this.

**Pull request merged**